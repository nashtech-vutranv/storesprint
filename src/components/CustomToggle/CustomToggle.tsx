import {FC, HTMLAttributes, useContext, useEffect, SetStateAction} from 'react'
import {AccordionContext, useAccordionButton} from 'react-bootstrap'
import {GlobalContext} from '../../store/GlobalContext'
import {AccordionActionType} from '../../store/actions'
import {AccordionName, IUrlParams} from '../../interface'

interface ICustomToggle extends HTMLAttributes<HTMLDivElement> {
  eventKey: string
  containerClass: string
  linkClass: string
  callback?: (eventKey: string) => void
  name: AccordionName
  selectedURLParamsObj?: IUrlParams
  setSelectedURLParamsObj?: (value: SetStateAction<IUrlParams>) => void
}

const CustomToggle: FC<ICustomToggle> = ({
  eventKey,
  containerClass,
  callback,
  style,
  name,
  selectedURLParamsObj,
  setSelectedURLParamsObj
}) => {
  const {state: {accordion}, dispatch: {
    accordion: accordionDispatch
  }} = useContext(GlobalContext)
  const {activeEventKey} = useContext(AccordionContext)
  const isExpand = activeEventKey === eventKey
  
  const decoratedOnClick = useAccordionButton(
    eventKey,
    () => callback && callback(eventKey)
  )

  useEffect(() => {
    if (activeEventKey) {
      setSelectedURLParamsObj && setSelectedURLParamsObj({
        ...selectedURLParamsObj,
        collapse: 'filter-accordion',
      })
      accordionDispatch({
        type: AccordionActionType.GET_CURRENT_ACCORDION_INFORMATION,
        payload: {
          ...accordion,
          [name]: activeEventKey
        }
      })
    } else {
      selectedURLParamsObj &&
        setSelectedURLParamsObj && setSelectedURLParamsObj({
          ...selectedURLParamsObj,
          collapse: '0',
        })
      accordionDispatch({
        type: AccordionActionType.GET_CURRENT_ACCORDION_INFORMATION,
        payload: {
          ...accordion,
          [name]: '0'
        }
      })
    }
  }, [activeEventKey])

  return (
    <span className={containerClass} style={style}>
      <span onClick={decoratedOnClick} style={{color: 'var(--ct-body-color)'}}>
        <div className='d-flex align-items-start'>
          {isExpand ? (
            <i
              className='mdi mdi-chevron-down d-none d-sm-inline-block align-middle'
              style={{transform: 'scale(1.5)'}}
            ></i>
          ) : (
            <i
              className='mdi mdi-chevron-right d-none d-sm-inline-block align-middle'
              style={{transform: 'scale(1.5)'}}
            ></i>
          )}
        </div>
      </span>
    </span>
  )
}

export default CustomToggle
