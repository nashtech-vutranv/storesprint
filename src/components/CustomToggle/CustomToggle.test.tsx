import {BrowserRouter as Router} from 'react-router-dom'
import {render, waitFor} from '@testing-library/react'
import '@testing-library/jest-dom'
import CustomToggle from './CustomToggle'

test('renders CustomToggle', async () => {
  const tree = render(
    <Router>
      <CustomToggle
        eventKey='test-key'
        containerClass='test-ctn'
        linkClass='test-lcl'
        name='relationship'
      />
    </Router>
  )
  await waitFor(() => {
    expect(tree).toMatchSnapshot()
  })
})
