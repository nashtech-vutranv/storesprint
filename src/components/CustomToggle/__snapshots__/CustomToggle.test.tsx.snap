// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CustomToggle 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="test-ctn"
      >
        <span>
          <div
            class="d-flex align-items-start"
          >
            <i
              class="mdi mdi-chevron-right d-none d-sm-inline-block align-middle"
              style="transform: scale(1.5);"
            />
          </div>
        </span>
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="test-ctn"
    >
      <span>
        <div
          class="d-flex align-items-start"
        >
          <i
            class="mdi mdi-chevron-right d-none d-sm-inline-block align-middle"
            style="transform: scale(1.5);"
          />
        </div>
      </span>
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
